src/compile/data/transforms.ts

```diff
@@ -45,6 +48,48 @@ export class FilterNode extends DataFlowNode {
   }
 }
 
+export class GeoJSONNode extends DataFlowNode {
+  public clone() {
+    return new GeoJSONNode(duplicate(this.fields), this.geojson, this.signal);
+  }
+
+  public static make(model: ModelWithField): GeoJSONNode {
+    const geo = model.reduceFieldDef((rel, fieldDef, channel) => {
+      if ((contains([X, Y], channel) && contains([LATITUDE, LONGITUDE], fieldDef.type))
```

Comment:
From the earlier ([collaped conversaton](https://github.com/vega/vega-lite/pull/2734#discussion_r133342790))
I'm confused.  If you say this doesn't make any sense, why do you generate it? 

Looking at `geo_layer.vg.json`, it seems like both geojson and geopoint are required though.